timer/timer.go

```diff
@@ -0,0 +1,68 @@
+// Package timer provides a simple timeout component.
+package timer
+
+import (
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+// TickMsg is a message that is sent on every timer tick.
+type TickMsg struct{}
+
+// TimeoutMsg is a message that is sent once when the timer times out.
+type TimeoutMsg struct{}
+
+// Model of the timer component.
+type Model struct {
+	// How long until the timer expires.
+	Timeout time.Duration
+
+	// How long to wait before every tick. Defaults to 1 second.
+	TickEvery time.Duration
```

Comment:
Actually—this should probably be `Interval` to match the other language, which is frankly better.